File: Backend/Microservices/Expense/EMS.Core/Entities/Common/AuditableEntity.cs
using EMS.Core.Common.Interfaces.Audit;

namespace EMS.Core.Entities.Common
{
    public class AuditableEntity : IAuditableEntity
    {
        public DateTimeOffset? CreatedAt { get; set; }
        public Guid? CreatedBy { get; set; }
        public DateTimeOffset? ModifiedAt { get; set; }
        public Guid? ModifiedBy { get; set; }
        public bool IsDeleted { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }
        public Guid? DeletedBy { get; set; }
    }
}

File: Backend/Microservices/Dispatcher/internal/models/common/auditable_entity.go
package common

import "time"

type AuditableEntity struct {
	CreatedAt time.Time `gorm:"column:CreatedAt"`
	CreatedBy string    `gorm:"column:CreatedBy"`
	UpdatedAt time.Time `gorm:"column:ModifiedAt"`
	UpdatedBy string    `gorm:"column:ModifiedBy"`
	IsDeleted bool      `gorm:"column:IsDeleted"`
	DeletedAt time.Time `gorm:"column:DeletedAt"`
	DeletedBy string    `gorm:"column:DeletedBy"`
}
